app/Http/Controllers/Api/AssetsController.php

```diff
@@ -736,5 +736,72 @@ public function audit(Request $request) {
 
 
 
+    }
+
+
+
+    /**
+     * Returns JSON listing of all assets
+     *
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * @param int $assetId
+     * @since [v4.0]
+     * @return JsonResponse
+     */
+    public function requestable(Request $request)
+    {
+
+
+        $allowed_columns = [
+            'id',
+            'name',
+            'asset_tag',
+            'serial',
+            'model_number',
+            'last_checkout',
+            'notes',
+            'expected_checkin',
+            'order_number',
+            'image',
+            'assigned_to',
+            'created_at',
+            'updated_at',
+            'purchase_date',
+            'purchase_cost',
+            'warranty_months',
+        ];
+
+
+        $assets = Company::scopeCompanyables(Asset::select('assets.*'),"company_id","assets")
+            ->with('location', 'assetstatus', 'assetlog', 'company', 'defaultLoc','assignedTo',
+                'model.category', 'model.manufacturer', 'model.fieldset','supplier')->where('assets.requestable', '=', '1');
+
+        $offset = request('offset', 0);
+        $limit = $request->input('limit', 50);
+        $order = $request->input('order') === 'asc' ? 'asc' : 'desc';
+        $assets->TextSearch($request->input('search'));
+
+        switch ($request->input('sort')) {
+            case 'model':
+                $assets->OrderModels($order);
+                break;
+            case 'model_number':
+                $assets->OrderModelNumber($order);
+                break;
+            case 'category':
+                $assets->OrderCategory($order);
+                break;
+            case 'manufacturer':
+                $assets->OrderManufacturer($order);
+                break;
+            default:
+                $assets->orderBy('assets.created_at', $order);
+                break;
+        }
+
+
+        $total = $assets->count();
+        $assets = $assets->skip($offset)->take($limit)->get();
+        return (new AssetsTransformer)->transformRequestedAssets($assets, $total);
```

Comment:
This entire method feels like copypasta to me - wouldn't you be able to get a list of `requestable` assets by calling the regular `index` method (above), but along with a `?requestable=true`? If so, we should do that. If not - is it possible to add any additional needed fields without breaking out a separate method?

Reply:
I wish I could see which method you were talking about here. Github. Sigh.

> wouldn't you be able to get a list of requestable assets by calling the regular index method (above), but along with a ?requestable=true

You can, but a list of assets requested by the user shouldn't have links and such back to the assets themselves, since most "regular" users won't have permissions to see them. This is a pared down listing with no objects, just strings, returned.

Reply:
So it was either create a new method, OR include some kind of a gross switch in the main Assets API Controller index.